enhancements/installer/install_ca_management.md

```diff
@@ -0,0 +1,164 @@
+  - "@abhinavdahiya"
+  - "@adambkaplan"
+  - "@dmage"
+approvers:
+  - "@sdodson"
+  - "@crawford"
+creation-date: 2020-08-03
+last-updated: 2020-08-03
+status: provisional
+---
+
+# Install CA Management
+
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Customers have a need to provide additional CAs to their cluster.  Ideally they can do this at 
+install time so that everything works by the time the cluster comes up (if they do not, the 
+cluster may install but things like imagestreams may not import, builds may not work, or other 
+components may not function as expected).
+
+Today the installer collects an “additionalTrustBundle”, and optionally proxy host names.
+
+The additionalTrustBundle is put(not sure where exactly?) and consumed directly by some set of 
+components such as the Machine Config Operator.  This ensures that the nodes can talk to the 
+mirror registry in a disconnected install for example.
+
+In addition if a proxy host is supplied, the bundle is put into a proxy configuration (CA+hosts) 
+which many/most components consume.
+
+However when only the bundle is supplied (because no proxy is in use) the bundle consumption is 
+essentially ad hoc. This has the negative consequence that components which don’t directly consume 
+it (from where?) may not function, particularly in a disconnected install where they need a CA to 
+trust the mirror registry.  We see this problem w/ imagestream import which expects to get its CAs 
+from an image config resource which references a configmap of bundles that are explicitly 
+associated to registry hostnames.  Further complicating the situation is that the field is simply 
+named `additionalTrustBundle` so the user intent is not clear, making it difficult for specific 
+components to know whether they should be consuming this value or not.
+
+The most direct consequences of this are that today if a customer does a disconnected install, 
+their imagestream imports from the mirror registry will fail.  In particular this means the 
+oauth-proxy imagestream, which is consumed by many OLM operators, will fail and thus those 
+operators will fail to function, until/unless the admin explicitly configures the CA in the image 
+config resource and then re-imports the imagestream.
+
+While this could theoretically be solved by the imagestream component trusting the additional CA (
+for all hosts?) explicitly, it would make more sense if we used the information the admin is 
+providing us at install time to correctly configure the cluster as if they had done it themselves, 
+rather than introducing additional sources of (potentially conflicting) configuration 
+information.
+
```

Comment:
A. The entire summary in general talks about how the installer currently configures 2 components with the CA provided to it, but there is another component imagestreams that also need to be configured with the CA. And the goals also match that we intend to support the imagestreams working correctly out of the box in local registry cases.

B. But there is text in the summary that is also talking about how currently the installer understands how to configure some components and it requires new components that need to configure the CA to update the installer to update the CA for them too... and this procedure is not sustainable? appropriate?

The summary and then the goals don't really line up imo. wdyt @bparees  ? Are we trying to say we should solve A with B but then we make B non-goal soon after below...

Reply:
> But there is text in the summary that is also talking about how currently the installer understands how to configure some components and it requires new components that need to configure the CA to update the installer to update the CA for them too... and this procedure is not sustainable? appropriate?

Sorry I couldn't determine what part of the summary you were referring to here.

What i was attempting to do in the summary is establish the precedence:  The installer currently configures CAs for certain components (namely the MCO and the ProxyConfig)

And then propose that therefore it should also configure CAs for additional components (namely the configuration needed for components that communicate w/ external registries).  Particularly since the MCO configuration is already there to enable communication w/ external registries, it's just not the first-class configuration mechanism we expect admins to use to enable communication w/ external registries.

Which non-goal appears to conflict w/ this?  I am guessing you are referring to this one:
> Adding generic cluster configuration fields to the installer that are not required to bootstrap the cluster and ensure consistent use of cluster config apis

It's probably a bit difficult to parse, but what I am trying to say, essentially, is, "there are two potential justifications for adding a field to the install-config.  One of them is because we need the info to bootstrap the cluster.  The other is we need the info to ensure that once the cluster is up, it is configured using the first-class configuration APIs an admin would expect and does not contain conflicting configuration information".

So i'm trying to limit the cases under which we would consider expanding the scope of the install-config generally, but this particular proposal falls into that latter category of allowing the installer to configure the cluster using the first class apis in a consistent way.  Thus it is not a violation of the non-goal.